lms/static/js/student_account/views/account_settings_factory.js

```diff
@@ -122,6 +73,82 @@
                 }
             ];
 
+            if (!isShibAuth) {
+                sectionsData[0].fields.push(
+                    {
+                        view: new FieldViews.TextFieldView({
+                            model: userAccountModel,
+                            title: gettext('Full Name'),
+                            valueAttribute: 'name',
+                            helpMessage: gettext('The name that appears on your Statements of Accomplishment. Other learners never see your full name.')
+                        })
+                    },
+                    {
+                        view: new AccountSettingsFieldViews.EmailFieldView({
+                            model: userAccountModel,
+                            title: gettext('Email Address'),
+                            valueAttribute: 'email',
+                            helpMessage: gettext('The email address you use to sign in to this site. Communications from us and your courses are sent to this address.')
+                        })
+                    },
+                    {
+                        view: new AccountSettingsFieldViews.PasswordFieldView({
+                            model: userAccountModel,
+                            title: gettext('Password'),
+                            screenReaderTitle: gettext('Reset your Password'),
+                            valueAttribute: 'password',
+                            emailAttribute: 'email',
+                            linkTitle: gettext('Reset Password'),
+                            linkHref: fieldsData.password.url,
+                            helpMessage: gettext('When you click "Reset Password", a message will be sent to your email address. Click the link in the message to reset your password.')
+                        })
+                    }
+                )
+            } else {
+                sectionsData[0].fields.push(
```

Comment:
since this is largely the same code as above, I wish it could be condensed and just use a variable to decide the type of field...